pkg/pod/entrypoint.go

```diff
@@ -84,9 +84,7 @@ var (
 // Containers must have Command specified; if the user didn't specify a
 // command, we must have fetched the image's ENTRYPOINT before calling this
 // method, using entrypoint_lookup.go.
-//
-// TODO(#1605): Also use entrypoint injection to order sidecar start/stop.
-func orderContainers(entrypointImage string, extraEntrypointArgs []string, steps []corev1.Container, results []v1beta1.TaskResult) (corev1.Container, []corev1.Container, error) {
+func orderContainers(entrypointImage string, commonExtraEntrypointArgs []string, extraEntrypointArgs [][]string, steps []corev1.Container, results []v1beta1.TaskResult) (corev1.Container, []corev1.Container, error) {
```

Comment:
Maybe instead of `extraEntrypointArgs`, which at least for now will always be a slice of at-most-two-element slices like `[]string{"-timeout", "30s"}`, we just make this `timeouts []*time.Duration`, where a `nil` means "no timeout"?

The addition of timeouts args, or extra args in general, feels like it belongs outside a method called `orderContainers`, which is so far primarily focused on setting `-wait_file`/`-post_file` flags. Should we rename the method to describe its new scope? Or split it into separate methods?